gems/sorbet-runtime/lib/types/private/final.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+# typed: false
+
+module T::Private::Final
+  module NoInherit
+    def inherited(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be inherited"
+    end
+  end
+
+  module NoIncludeExtend
+    def included(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be included"
+    end
+
+    def extended(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be extended"
+    end
+  end
+
+  def self.declare(mod)
+    if !mod.is_a?(Module)
+      raise "#{mod.name} is not a class, module or method and cannot be declared as final"
+    end
+    if final_module?(mod)
+      raise "#{mod.name} was already declared as final and cannot be re-declared as final"
+    end
+    if T::AbstractUtils.abstract_module?(mod)
+      raise "#{mod.name} was already declared as abstract and cannot be declared as final"
+    end
+    mod.extend(mod.is_a?(Class) ? NoInherit : NoIncludeExtend)
```

Comment:
Is there a reason why we stopped using Chalk::Tools.replace_method?

Reply:
Not particularly, I think either would work. Should I prefer replace_method?

Reply:
I guess it's fine either way.